examples/LayoutSpecExamples/Sample/LayoutExampleNodes.m

```diff
@@ -0,0 +1,261 @@
+    _postLocationNode.maximumNumberOfLines = 1;
+    _postLocationNode.truncationMode = NSLineBreakByTruncatingTail;
+  }
+  
+  return self;
+}
+
+- (ASLayoutSpec *)layoutSpecThatFits:(ASSizeRange)constrainedSize
+{
+  // when the username / location text is too long, shrink the stack to fit onscreen rather than push content to the right, offscreen
+  ASStackLayoutSpec *nameLocationStack = [ASStackLayoutSpec verticalStackLayoutSpec];
+  nameLocationStack.style.flexShrink = YES;
+  nameLocationStack.style.flexGrow = YES;
+  
+  // if fetching post location data from server, check if it is available yet and include it if so
+  if (_postLocationNode.attributedText) {
+    nameLocationStack.children = @[_usernameNode, _postLocationNode];
+  } else {
+    nameLocationStack.children = @[_usernameNode];
+  }
+  
+  // horizontal stack
+  ASStackLayoutSpec *headerStackSpec = [ASStackLayoutSpec stackLayoutSpecWithDirection:ASStackLayoutDirectionHorizontal
+                                                                               spacing:40
+                                                                        justifyContent:ASStackLayoutJustifyContentStart
+                                                                            alignItems:ASStackLayoutAlignItemsCenter
+                                                                              children:@[nameLocationStack, _postTimeNode]];
+  
+  // inset the horizontal stack
+  return [ASInsetLayoutSpec insetLayoutSpecWithInsets:UIEdgeInsetsMake(0, 10, 0, 10) child:headerStackSpec];
+}
+
+@end
+
+
+@implementation PhotoWithInsetTextOverlay
+
++ (NSString *)title
+{
+  return @"Photo with inset text overlay";
+}
+
++ (NSString *)descriptionTitle
+{
+  return @"try rotating me!";
+}
+
+- (instancetype)init
+{
+  self = [super init];
+  
+  if (self) {
+    self.backgroundColor = [UIColor clearColor];
+    
+    _photoNode = [[ASNetworkImageNode alloc] init];
+    _photoNode.URL = [NSURL URLWithString:@"http://asyncdisplaykit.org/static/images/layout-examples-photo-with-inset-text-overlay-photo.png"];
+    _photoNode.willDisplayNodeContentWithRenderingContext = ^(CGContextRef context) {
+      CGRect bounds = CGContextGetClipBoundingBox(context);
+      [[UIBezierPath bezierPathWithRoundedRect:bounds cornerRadius:10] addClip];
+    };
+    
+    _titleNode = [[ASTextNode alloc] init];
+    _titleNode.maximumNumberOfLines = 2;
+    _titleNode.truncationMode = NSLineBreakByTruncatingTail;
+    _titleNode.truncationAttributedText = [NSAttributedString attributedStringWithString:@"..." fontSize:16 color:[UIColor whiteColor]];
+    _titleNode.attributedText = [NSAttributedString attributedStringWithString:@"family fall hikes" fontSize:16 color:[UIColor whiteColor]];
+  }
+  
+  return self;
+}
+
+- (ASLayoutSpec *)layoutSpecThatFits:(ASSizeRange)constrainedSize
+{
+  _photoNode.style.preferredSize = CGSizeMake(constrainedSize.max.width / 4.0, constrainedSize.max.width / 4.0);
```

Comment:
As it seems like width and height has the same calculation, should we move that to a temporary variable above?